pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -132,9 +133,13 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 }
 
 func (pc *PodCache) labelFilter(old, cur *v1.Pod) bool {
-	// If labels updated, trigger proxy push
-	if cur.Status.PodIP != "" && !maps.Equal(old.Labels, cur.Labels) {
-		pc.proxyUpdates(cur.Status.PodIP)
+	// If labels/annotations updated, trigger proxy push
+	labelsChanged := !maps.Equal(old.Labels, cur.Labels)
+	// Annotations are only used in endpoints in one case, so just compare that one
+	relevantAnnotationsChanged := old.Annotations[constants.AmbientRedirection] != cur.Annotations[constants.AmbientRedirection]
```

Comment:
Oh this is why you wanted to ditch the annotation.

Can you throw a TODO in here and linkref to https://github.com/istio/istio/issues/50355?

Reply:
This actually isn't related to my discussion around annotation actually -- even if we do that, we still need this IMO. For instance, if I enroll the entire namespace, then the CNI will annotate the pod even with that proposal I think?

Reply:
I'm saying we could make that annotation the CNI does be a label (which that issue discusses as an option), and we then wouldn't need to check annotations JUST for ambient status.

Reply:
Ah ok, at one point it had been discussed "humans use labels, machines (CNI) use annotations". I don't mind the annotation check here really -- I wouldn't weigh it too much on #50355